### ISSUE TITLE
How to configure API?

### ISSUE THREAD

**Issue by KrzysztofMadejski** [code4romania/monitorizare-vot#51]
After reading https://github.com/code4romania/monitorizare-vot/blob/translate/entities/GettingStarted.md I'm still unsure how to configure the API. Few questions that hopefully will clarify that:

- [ ] Is [appsettings.json](https://github.com/code4romania/monitorizare-vot/blob/master/private-api/app/src/VotingIrregularities.Api/appsettings.json)  default settings or is it just a template that needs to be overriden?
  - if these are default settings then it should not contain `"SecretKey": "SuperSecretKetyTBRSuperSecretKetyTBRSup...`, it should just be empty and set elsewhere; the same goes with all `TBR` strings
  - if this is a template then it should be named IMHO `appsettings.template.json` because 1) it's more intuitional and 2) I've read that `appsettings.json` is the default config filename so it will be automatically read by .NET

- [ ] Can .NET read the settings from environment or should they be coded in appsettings like it is now? It would be easier to set them in environment and somehow reference in code. Now a deployment script need to rewrite the settings
    ```
    envsubst "`printf '${%s} ' $(bash -c "compgen -A variable")`" < appsettings.template.json > appsettings.json`
    ```

**Comment by KrzysztofMadejski** [comment]
I see that piping in environmental variables is quite easy: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/index?view=aspnetcore-2.1#environment-variables-configuration-provider

**Comment by KrzysztofMadejski** [comment]
We've been talking about it in Romania and we agreed that `appsettings.json` file is the default settings, that can be overriden by specific environment files, such as `appsettings.production.json` or the environment variables if such behaviour is code in the startup.

So I propose to resolve it as follows:
- [ ] limit `appsettings.json` to contain only the default settings 
- [ ] create `appsettings.yourenvironment.json` as a template file with all the settings that need to be set up (db, salt, etc.)
- [ ] add line of code that would read settings from environment variables
- [ ] in docker use environment variables to set everything up instead of substituting values in the file

I commit to do it over coming weeks. This should be transparent from existing deployments perspective.

**Comment by aniri** [comment]
closing at it will be fixed in #100
